connmgr/connmanager.go

```diff
@@ -254,6 +259,42 @@ func (cm *ConnManager) handleFailedConn(c *ConnReq, err error) {
 	}
 }
 
+func (cm *ConnManager) releaseAddress(addr *net.TCPAddr) {
+	cm.addressMtx.Lock()
+	defer cm.addressMtx.Unlock()
```

Comment:
Please add empty line after defer, to create a visual boundary between locks/unlocks and the actual code